pandas/io/formats/xml.py

```diff
@@ -0,0 +1,601 @@
+        """
+
+        nmsp_dict: Dict[str, str] = {}
+        if self.namespaces and self.prefix is None:
+            nmsp_dict = {"xmlns": n for p, n in self.namespaces.items() if p != ""}
+
+        if self.namespaces and self.prefix:
+            nmsp_dict = {"xmlns": n for p, n in self.namespaces.items() if p == ""}
+
+        return nmsp_dict
+
+    def build_attribs(self) -> None:
+        """
+        Create attributes of row.
+
+        This method adds attributes using attr_cols to row element and
+        works with tuples for multindex or hierarchical columns.
+        """
+
+        raise AbstractMethodError(self)
+
+    def build_elems(self) -> None:
+        """
+        Create child elements of row.
+
+        This method adds child elements using elem_cols to row element and
+        works with tuples for multindex or hierarchical columns.
+        """
+
+        raise AbstractMethodError(self)
+
+    def write_output(self) -> Optional[str]:
+        xml_doc = self.build_tree()
+        out_str: Optional[str] = xml_doc.decode(self.encoding).rstrip()
+
+        try:
+            if self.path_or_buffer and isinstance(self.path_or_buffer, str):
+                with open(self.path_or_buffer, "wb") as f:
+                    f.write(xml_doc)
+
+                out_str = None
+
+        except (UnicodeDecodeError, OSError, FileNotFoundError) as e:
+            raise e
+
+        return out_str
+
+
+class EtreeXMLFormatter(BaseXMLFormatter):
+    """
+    Class for formatting data in xml using Python standard library
+    modules: `xml.etree.ElementTree` and `xml.dom.minidom`.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.validate_columns()
+        self.validate_encoding()
+        self.handle_indexes()
+        self.prefix_uri = self.get_prefix_uri()
+
+    def build_tree(self) -> bytes:
+        from xml.etree.ElementTree import Element, SubElement, tostring
+
+        self.root = Element(
+            f"{self.prefix_uri}{self.root_name}", attrib=self.other_namespaces()
+        )
+
+        for k, d in self.frame_dicts.items():
+            self.d = d
+            self.elem_row = SubElement(self.root, f"{self.prefix_uri}{self.row_name}")
+
+            if not self.attr_cols and not self.elem_cols:
+                self.elem_cols = list(self.frame_dicts[0].keys())
+                self.build_elems()
+            else:
+                self.build_attribs()
+                self.build_elems()
+
+        self.out_xml = tostring(self.root, method="xml", encoding=self.encoding)
+
+        if self.pretty_print:
+            self.out_xml = self.prettify_tree()
+
+        if not self.xml_declaration:
+            self.out_xml = self.remove_declaration()
+
+        if self.stylesheet:
+            warn(
+                "To use stylesheet, you need lxml installed. "
+                "Instead, the non-transformed, original XML is returned.",
+                UserWarning,
```

Comment:
why is this not an exception?

Reply:
Changing to `ValueError` in read and write scripts for wrong use of `stylesheet` with etree parser. And added tests.